### PR TITLE
Automatically generate Helm chart documentation

### PR DESCRIPTION
**Pull request by adutra**
This commit introduces the usage of
https://github.com/norwoodj/helm-docs
to generate documentation for the Nessie
Helm chart.

It also modifies the release workflow
so that it executes helm-docs on the updated
chart.

### CODE REVIEW

File: .github/workflows/release-create.yml [hunk 1]
```diff
@@ -184,6 +186,11 @@ jobs:
         cp /tmp/HISTORY.rst python/HISTORY.rst
         rm /tmp/HISTORY.rst
 
+    - name: Update helm/README.md
+      run: |
+        go install github.com/norwoodj/helm-docs/cmd/helm-docs@latest
+        $(go env GOPATH)/bin/helm-docs --chart-search-root=helm
```
Comment (adutra) [inline .github/workflows/release-create.yml:192]: This was manually tested on my fork.

**Comment by adutra** [comment]
Please note: helm-docs is distributed under GNU General Public License v3.0. I think that's fine as long as we are only executing the binary during the build, not redistributing it in any way.

**Review by dimas-b — COMMENTED** [review]
Nice improvement :) Minor comments below.

File: helm/nessie/values.yaml [hunk 2]
```diff
@@ -1,138 +1,181 @@
...
-  #  allowViewingBranch: op=='VIEW_REFERENCE' && role.startsWith('test_user') && ref.startsWith('allowedBranch')
-  #  allowCommits: op=='COMMIT_CHANGE_AGAINST_REFERENCE' && role.startsWith('test_user') && ref.startsWith('allowedBranch')
+  # -- The authorization rules when authorization.enabled=true. Example rules can be found at https://projectnessie.org/features/metadata_authorization/#authorization-rules
+  rules:
+    {}
+    # allowViewingBranch: op=='VIEW_REFERENCE' && role.startsWith('test_user') && ref.startsWith('allowedBranch')
+    # allowCommits: op=='COMMIT_CHANGE_AGAINST_REFERENCE' && role.startsWith('test_user') && ref.startsWith('allowedBranch')
 
-# configure jaeger tracing for the nessie server
 jaegerTracing:
```
[the remark this answers is not in the document]
Comment (adutra) [reply helm/nessie/values.yaml:95]: Indeed, Jaeger and OpenTracing are now considered [deprecated technologies](https://quarkus.io/guides/opentracing) by Quarkus, so we probably should deprecate/remove this section from the Helm chart. I will create a follow-up issue if you like.

File: helm/nessie/values.yaml [hunk 3]
```diff
@@ -1,138 +1,181 @@
...
 
-# this is needed when selecting ROCKS
+# RocksDB settings. Only required when using ROCKS version store type; ignored otherwise.
 rocksdb:
+  # -- The storage class name of the persistent volume claim to create.
   storageClassName: standard
+  # -- The size of the persistent volume claim to create.
   storageSize: 1Gi
+  # -- The path to the rocksdb database.
   dbPath: /rocks-nessie
```
[the remark this answers is not in the document]
Comment (adutra) [reply helm/nessie/values.yaml:26]: Good point. I will remove the docs for this setting in this PR, then create a follow-up PR to remove it completely. Does that sound OK to you?

File: helm/nessie/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply helm/nessie/values.yaml:95]: Yes, please :)

File: helm/nessie/values.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (adutra) [reply helm/nessie/values.yaml:27]: Good point, I will remove the docs in this PR and open a PR to remove it later.

**Review by dimas-b — APPROVED** [review]

File: helm/nessie/values.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (adutra) [reply helm/nessie/values.yaml:27]: https://github.com/projectnessie/nessie/issues/6104

File: helm/nessie/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (adutra) [reply helm/nessie/values.yaml:100]: https://github.com/projectnessie/nessie/issues/6105

**Comment by snazy** [comment]
Could the CI failures on main be related to this change?

**Comment by dimas-b** [comment]
Should we lint helm on PRs (at least under a label perhaps)?

**Comment by dimas-b** [comment]
Yeah, we already have `pr-helm`... should have used it :facepalm:

**Comment by adutra** [comment]
I will provide a fix, sorry for that.

**Pull request merged**